ipam/allocator.go

```diff
@@ -115,6 +117,41 @@ func (a *Allocator) updateBitMasks(aSpace *addrSpace) error {
 	return nil
 }
 
+// Fixes damaged bitmask. Meant to be called in constructor only.
+// Check could be positive during docker upgrade from 1.9.1 to newer versions
+func (a *Allocator) checkConsistency(as string) {
+	// List of affected master pools
+	pdList := map[string]*PoolData{}
+
+	// Retrieve current configuration and bitmask data from
+	// the datastore about this address space
+	a.refresh(as)
+	aSpace := a.addrSpaces[as]
+	a.updateBitMasks(aSpace)
+
+	// Look for corrupted bitmask data
+	for sk, pd := range aSpace.subnets {
+		if pd.Range != nil {
+			continue
+		}
+		bm := a.addresses[sk]
+		if bm.IsCorrupted() {
+			log.Debugf("Detected corrupted bitmask: %s. Fixing it.", bm)
+			pdList[sk.String()] = pd
+		}
+	}
+
+	// Re install the affected address pools
+	for sk, pd := range pdList {
+		if err := a.ReleasePool(sk); err != nil {
```

Comment:
I was thinking more about the `globalAddressSpace` in which case we cannot nuke it like this and so if we have some way to fix the bitmask in place then even if two hosts try to fix it we should be able to do it in a race-free manner